tests/test_writing.py

```diff
@@ -35,18 +35,6 @@ def validate_file(self, path: str, object_type: str) -> List[Any]:
     def validate_link_types(
         self, root_href: str, catalog_type: pystac.CatalogType
     ) -> None:
-        def validate_asset_href_type(item: pystac.Item, item_href: str) -> None:
-            for asset in item.assets.values():
-                if not is_absolute_href(asset.href):
-                    is_valid = not is_absolute_href(asset.href)
-                    if not is_valid:
-                        # If the item href and asset href don't share
-                        # the same root, the asset href must be absolute
-                        rel_href = make_relative_href(asset.href, item_href)
-                        self.assertEqual(asset.href, rel_href)
-                    else:
-                        self.assertTrue(is_valid)
-
```

Comment:
Looking back through the commit history it seems like this test did have a purpose originally, but that it's been degraded over time. @lossyrob Do you know what the original intent was?

Reply:
Yep, it used to be that when you saved a catalog as SELF_CONTAINED and/or RELATIVE_PUBLISHED it would also modify the asset hrefs to be absolute/relative (I forget which). Looks like some of the catalog type information got deleted here, probably as part of the change to remove LinkType. So this looks like code that was refactored to pass with refactors, but ended up meaningless.